### PR TITLE
New HelloWorldExample [12385]

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Simple HelloWorld example with a monitor listening for FastDDS statistics and printing out the latency and throughput of the communication between publisher and subscriber.

### CODE REVIEW

File: examples/cpp/HelloWorldExample/arg_configuration.h [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        return option::ARG_ILLEGAL;
+    }
+
+};
+
+enum optionIndex
+{
+    UNKNOWN_OPT,
+    HELP,
+    WAIT,
```
Comment (jparisu) [inline examples/cpp/HelloWorldExample/arg_configuration.h:108]: I would totally remove this argument

File: examples/cpp/HelloWorldExample/Monitor.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+                  << publication_throughput.second << " B/s]" << std::endl;
+    }
+
+    return publication_throughput_data;
+}
+
+/***************************************************************
+* Monitor Listener callbacks implementation
+***************************************************************/
+void Monitor::Listener::on_participant_discovery(
```
Comment (jparisu) [inline examples/cpp/HelloWorldExample/Monitor.cpp:190]: **In general** 
I would do specific discovery messages for each entity kind
The `id` here is not relevant to the user, but more like a mess. Add GUID in Participant, Writer and Reader, comment that a new topic has been discovered, mention the physical data when it arrives. In summary, use `get_info` with the new entities and give some information

**Review by jparisu — DISMISSED** [review]
LGTM 
uncrustify fails in autogenerated files
test fails in leaky test case

**Comment by EduPonz** [comment]
@juanlofer-eprosima Can you please rebase this?

**Review by jparisu — CHANGES_REQUESTED** [review]
LGTM

Please, fix uncrustify

**Review by jparisu — DISMISSED** [review]
LGTM 

Uncrustify fails because autogenerated or thirdparty files

**Pull request merged**